### PR TITLE
Loggers refactoring

### PR DESCRIPTION
**Pull request by ataymano**
Please do not merge yet - half done loggers refactoring + events sampling. Want to use this review as a reference in tomorrow's discussion.

### CODE REVIEW

File: reinforcement_learning/rlclientlib/logger/event_hub_logger.h [hunk 1]
```diff
@@ -9,68 +10,72 @@
 #include "api_status.h"
 #include "../error_callback_fn.h"
 #include "utility/watchdog.h"
+#include "ranking_response.h"
+#include "ranking_event.h"
 
 namespace reinforcement_learning {
   // This class wraps logging event to event_hub in a generic way that live_model can consume.
-  class event_hub_logger : public i_logger {
+  class event_hub_logger {
```
Comment (rajan-chari) [inline reinforcement_learning/rlclientlib/logger/event_hub_logger.h]: >event_hub_logger [](start = 8, length = 16)

Should be renamed without event_hub as suggested by Alexey.

**Comment by jackgerrits** [comment]
I would like #1585 to be merged before this if possible as it was added to verify logger functionality and this is touching that section

**Comment by rajan-chari** [comment]
Got it.

---
In reply to: [419948622](https://github.com/JohnLangford/vowpal_wabbit/pull/1594#issuecomment-419948622) [](ancestors = 419948622)

File: reinforcement_learning/rlclientlib/logger/event_hub_logger.h [hunk 2]
```diff
@@ -9,68 +10,72 @@
 #include "api_status.h"
 #include "../error_callback_fn.h"
 #include "utility/watchdog.h"
+#include "ranking_response.h"
+#include "ranking_event.h"
 
 namespace reinforcement_learning {
   // This class wraps logging event to event_hub in a generic way that live_model can consume.
-  class event_hub_logger : public i_logger {
```
Comment (rajan-chari) [inline reinforcement_learning/rlclientlib/logger/event_hub_logger.h]: >event_hub_logger [](start = 8, length = 16)

Placeholder:  Wait for changes discussed offline.  (name changes and moving periodic_proc to event_logger.

**Comment by rajan-chari** [comment]
Looks good.

**Pull request merged**